src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -137,13 +139,56 @@
     (when platform/android?
       (>evt [::mentions/calculate-suggestions mentionable-users]))))
 
+(defn force-text-input-update!
+  "force-text-input-update! forces the
+  input to re-render, necessary when we are setting value"
+  []
+  (swap! chat-input-key inc))
+
+(re-frame/reg-fx
+ :set-input-text
+ (fn [[chat-id text]]
+    ;; We enable mentions
+   (swap! mentions-enabled assoc chat-id true)
+   (on-text-change text chat-id)
+    ;; We update the key so that we force a refresh of the text input, as those
+    ;; are not ratoms
+   (force-text-input-update!)))
+
+(fx/defn set-input-text
```

Comment:
There's quite a bunch of important things happening in this event handler. To paraphrase the re-frame docs:

> Event Handlers - most of your testing focus will be here because this is where most of the logic lives
> Subscription Handlers - often not a lot to test here. Only [Layer 3](https://github.com/day8/re-frame/blob/master/docs/subscriptions.md) subscriptions need testing.

In the end, I think we should strive to at least write tests for things that touch the app db, especially for non-trivial event handlers like this one. Another way of helping with testability of this event handler is to extract the data processing logic to a separate function and unit test it.

Reply:
Yeah, actually I just moved this event handler from an old component to the new one, and I need a bit more understanding of why (and if) this is needed, because it has mostly logic for mentions (which is not part of this issue, so maybe it could be simplified, but IDK if this is something needed for mentions in the redesign or if the logic will change, so probably we don't want to spend extra energy on figuring this out at this stage / PR).

Currently what matters for this fix is the `set-input-text` `fx` which forces the text input to update, so I removed that event handler and update the code to trigger it from other event handler (`:chat.ui/edit-message`).

If we need the mentions logic we can add it and refactor it, need to check if we have issues created and designs for mentions. I will create them if we don't have them.

Reply:
Okay, I understand the context now @briansztamfater, and thanks for creating related issues o/

May I give a suggestion to help reviewers understand the scope of the PR? If the code being added is mostly a copy of something, we can let reviewers know by adding an FYI Github comment or just a quick explanation on the PR description. This way reviewers can better judge when to suggest improvements.

Reply:
Sure, I'll add a note on the PR in following similar cases. Thanks for the suggestion!